torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -211,6 +211,18 @@ Tensor mvlgamma_backward(Tensor grad, const Tensor & self, int64_t p) {
   return grad * args.digamma_().sum(-1);
 }
 
+Tensor mul_scalar_backward(Tensor grad, Tensor self, Scalar other) {
+  auto scalar_as_tensor = at::scalar_to_tensor(other);
+  auto result = grad * scalar_as_tensor.conj();
+  // only real valued gradient should be propagated for real tensors
+  if (!self.is_complex() && result.is_complex()) {
+    // R -> C
+    result = at::real(result);
```

Comment:
Why is that the right thing to do?
Why not the sum of the real and imaginary part?

Reply:
Answer: The trick is that we implement the backward formula as if y was provided and was 0. And then take the real part of it. This gives the same result as if we derived the formula directly but it is simpler to write the overall code this way.